### DISCUSSION TITLE
contributing/development/configuring_an_ide/clion

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#116] [User-contributed notes]
# contributing/development/configuring_an_ide/clion

CLion is a commercial JetBrains IDE for C++. Importing the project: CLion can import a project's compilation database file, commonly named compile_commands.json. To generate the compilation databas...

https://docs.godotengine.org/en/stable/contributing/development/configuring_an_ide/clion.html

**Comment by ajpgarner** [comment]
Using CLion 2023.3.4 on Windows 10, I found it necessary to explicitly choose specifically the `.console.exe` version of the editor, as otherwise I had the aforementioned issue of it only being possible to debug the Godot Project Manager window.

**Reply by JamesWIToll** [reply]
Hi, did you get breakpoints working with this method? When I specifically choose the .console.exe version my breakpoints aren't hit and I get the message:
'''
The breakpoint will not currently be hit. No executable code is associated with this line.
'''

**Comment by nmerget** [comment]
Using CLion 2024.2.1 on Windows 11, to enable the debugger I had to go to `Settings` -> `Build, Execution, Deployment` -> `Toolchain`: Select `MinGW (Default)` and move it down, to make`Visual Studio` the default. 

Otherwise I got the error at all breakpoints: `The breakpoint will not currently be hit. No executable code is associated with this line`

**Reply by voxelv** [reply]
This didn't fix it for me. The problem I had was that I selected the wrong executable. 
I had selected 
`godot.windows.editor.x86_64.exe`
but I should have selected
`godot.windows.editor.dev.x86_64.exe`
(Note the `dev` in the second one)

**Reply by BadMachine** [reply]
thanks brav

**Comment by Nodragem** [comment]
On my side, I get a warning `This file does not belong to any project target, code insight features might not work`, any ideas how to fix it? I think I need it to clean my `#include` list.

**Reply by nmerget** [reply]
Three things I found out (for windows):

1. I had to do was to add `cl` to my `PATH` like [this](https://github.com/godotengine/godot-docs/issues/3311#issuecomment-1364430566).

2. You need to run `scons compiledb=yes test=yes` if you want to add all files inside `tests` directories into the `compile_commands.json`. 

3. If you create a new file you need to run  `scons compiledb=yes` again. Maybe you can copy paste a block from `compile_commands.json` and just change the path, but anywas you need to build the `.obj` file via `scons`

**Comment by stan220** [comment]
Hi, after run `Debug` or `Attach to process` I get a message like this:
```
process exited with status -1 (attach failed (Not allowed to attach to process.  Look in the console messages (Console.app), near the debugserver entries, when the attach failed.  The subsystem that denied the attach permission will likely have logged an informative message about why it was denied.))
```

I tried search how to fix this error, but I couldn't find how to allow permission to attach on mac.

My setup:
```
Godot Engine v4.3.stable.official.77dcf97d8 - https://godotengine.org
Vulkan 1.2.283 - Forward+ - Using Device #0: Apple - Apple M3 Pro
```

Could you help me please?

**Reply by Calinou** [reply]
Please ask support questions on other community platforms: https://godotengine.org/community/

**Comment by Toxe** [comment]
> Give the tool a name, e.g. `Build Godot debug`, set **Program** to `scons` [...]

I installed Scons via Scoop and I had to set the program to `C:\Users\MY_USER_NAME\scoop\shims\scons.cmd`.

**Comment by jmielc2** [comment]
> Then, open the Godot root directory with CLion. CLion will import the compilation database, index the codebase, and provide autocompletion and other advanced code navigation and refactoring functionality.

This did not happen for me. To enable IntelliSense features, I had to find the `CMakeLists.txt` file located in the `platform/android/java/nativeSrcsConfigs` directory, right click, and select `Load CMake Project` for the IDE to properly index the repo. This added an additional `godot` build configuration which I simply deleted.
